cmd/kubelet/app/server.go

```diff
@@ -696,7 +697,30 @@ func RunKubelet(kubeFlags *options.KubeletFlags, kubeCfg *kubeletconfiginternal.
 		kubeDeps.OSInterface = kubecontainer.RealOS{}
 	}
 
-	k, err := builder(kubeCfg, kubeDeps, &kubeFlags.ContainerRuntimeOptions, kubeFlags.HostnameOverride, kubeFlags.NodeIP, kubeFlags.ProviderID, kubeFlags.CloudProvider, kubeFlags.CertDirectory, kubeFlags.RootDirectory)
+	k, err := builder(kubeCfg,
+		kubeDeps,
+		&kubeFlags.ContainerRuntimeOptions,
```

Comment:
holy crap, can't we just pass the whole flags?

Reply:
People don't like it if pkg depends on cmd, which it would have to to get the type for the flags struct. Doesn't necessarily make sense to move the flags struct to pkg if it contains stuff that's only used in cmd.